app/views/step_by_step_pages/schedule.html.erb

```diff
@@ -60,10 +60,28 @@
         ]
       } %>
 
-      <div class="form-group">
-        <%= label_tag "schedule[time]", 'Time' %>
-        <%= text_field_tag "schedule[time]" %>
-      </div>
+      <% time_errors = @step_by_step_page.issues.include?(:schedule_time) ? [
+        {
+          text: @step_by_step_page.issues[:schedule_time],
+        }
+      ] : false %>
+      <%= render "components/autocomplete", {
+        id: "time",
+        name: "schedule[time]",
+        label: {
+          text: t("schedule.edit.time"),
+          bold: true,
+        },
+        error_items: time_errors,
+        input: {
+          value: params.dig(:schedule, :time) || publish_time.strftime("%-l:%M%P"),
```

Comment:
Content Publisher has a helper for formatting the time nicely. I think we can add that in a future PR and at the same time change the datetime formatting on the publish-or-delete screen for scheduled step by steps. Think it's out of scope for this PR.

Reply:
What is this check doing? What is the difference between `params` and `publish_time`.

Can this logic be moved out of the view?

Reply:
For example, why is `publish_time` being formatted with `("%-l:%M%P")`? What does that format look like?

Reply:
Have now moved this logic out of the view and moved the formatting to a helper method with tests 👍